scanner/src/main/kotlin/scanners/ScanCode.kt

```diff
@@ -265,4 +258,101 @@ object ScanCode : LocalScanner() {
 
         return onlyTimeoutErrors
     }
+
+    /**
+     * Get the SPDX license id (or a fallback) for a license finding.
+     */
+    private fun getLicenseId(license: JsonNode): String {
+        var name = license["spdx_license_key"].asText()
+
+        if (name.isEmpty()) {
+            val key = license["key"].asText()
+            name = if (key == "unknown") "NOASSERTION" else "LicenseRef-$key"
+        }
+
+        return name
+    }
+
+    /**
+     * Get the license found in one of the commonly named license files, if any, or an empty string otherwise.
+     */
+    internal fun getRootLicense(result: JsonNode): String {
+        val matchersForLicenseFiles = LICENSE_FILE_NAMES.map {
+            FileSystems.getDefault().getPathMatcher("glob:$it")
+        }
+
+        val rootLicenseFile = result["files"].singleOrNull {
+            val path = it["path"].asText()
+            matchersForLicenseFiles.any { it.matches(Paths.get(path)) }
+        } ?: return ""
+
+        val rootLicense = rootLicenseFile["licenses"].singleOrNull() ?: return ""
+
+        return getLicenseId(rootLicense)
+    }
+
+    internal fun getClosestCopyrightStatements(copyrights: JsonNode, startLine: Int, tolerance: Int = 5):
+            SortedSet<String> {
+        val closestCopyrights = copyrights.filter {
+            Math.abs(it["start_line"].asInt() - startLine) <= tolerance
```

Comment:
You could use `(it["start_line"].asInt() - startLine).absoluteValue`. Not shorter, but more Kotlin :).

Reply:
Done.